pkg/cache/internal/informers.go

```diff
@@ -250,7 +266,12 @@ func (ip *Informers) Get(ctx context.Context, gvk schema.GroupVersionKind, obj r
 		}
 	}
 
-	if started && !i.Informer.HasSynced() {
+	shouldBlock := true
+	if cfg.BlockUntilSynced != nil {
+		shouldBlock = *cfg.BlockUntilSynced
+	}
+
+	if started && !i.Informer.HasSynced() && shouldBlock {
```

Comment:
```suggestion
	if shouldBlock && started && !i.Informer.HasSynced() {
```
nit: minor optimization but should avoid the call to HasSynced if shouldBlock is false